subgraphs/arbitrum-one-bridge/src/handlers/ethereum/messagesAndEthDeposits.ts

```diff
@@ -0,0 +1,76 @@
+import { SDK } from "../../sdk/protocols/bridge";
+import {
+  BridgePoolType,
+  CrosschainTokenType,
+} from "../../sdk/protocols/bridge/enums";
+import { Address } from "@graphprotocol/graph-ts";
+import { networkToChainID } from "../../sdk/protocols/bridge/chainIds";
+import { ETH_ADDRESS, ETH_SYMBOL, Network } from "../../sdk/util/constants";
+import { MessageDelivered } from "../../../generated/L1Bridge/Bridge";
+import { ethSideConf, Pricer, TokenInit } from "../../common/utils";
+
+// eth
+const ethAddress = Address.fromString(ETH_ADDRESS);
+
+export function handleL1MessageDelivered(event: MessageDelivered): void {
+  // -- SDK
+
+  const sdk = SDK.initialize(ethSideConf, new Pricer(), new TokenInit(), event);
+
+  // -- ACCOUNT
+
+  const acc = sdk.Accounts.loadAccount(event.params.sender);
+
+  // -- HANDLE ETH DEPOSIT & MESSAGES
+
+  // Nitro
+  // Message Types - https://github.com/OffchainLabs/nitro/blob/master/contracts/src/libraries/MessageTypes.sol#L10
+  // L1MessageType_ethDeposit = 12
+  // L2_MSG = 3
+
+  if (event.params.kind == 12) {
```

Comment:
Can you put this `12` into a named constant? Same for the `3` in the next if branch.

Also, are the rest of message types safe to ignore for our purpose?

Reply:
Will do for the named constants.

> Also, are the rest of message types safe to ignore for our purpose?
I think so but I’ll review this again. Here’s the full list of [L2 messages](https://github.com/OffchainLabs/nitro/blob/master/contracts/src/libraries/MessageTypes.sol#L10).

@jaimehgb  Can you define again what kind of message types should we consider for tracking under the bridge? all types including ones required for L2 operations vs generic types vs deposit money, etc.

Reply:
@jaimehgb I believe we should ignore the other message types because they are related to protocol operations or sending transactions vs messages (which are tracked using `L2_MSG` event and are sent by invoking `sendL2Message()`, `sendL2MessageFromOrigin()`). 

See [Arbitrum’s Inbox]( https://github.com/OffchainLabs/nitro/blob/master/contracts/src/bridge/Inbox.sol) for more details.

Reply:
Makes sense 👍